docs/docs/migration-guide.mdx

```diff
@@ -126,6 +126,152 @@ confidences. This change can affect the maximum confidence predicted by the
 However, the previous behavior can still be enforced by setting
 `ranking_length=10` and `renormalize_confidences=True`.
 
+### Slot Mappings
+
+As of version 3.0, there is a single explicit mechanism for slot filling enabled by defining slot mappings for each slot
+in the `slots` section of the domain file. This approach keeps slots up to date over the course of a conversation, and
+removes duplicated effort in mapping the same slots in multiple forms.
+
+This new mechanism replaces the implicit slot setting via auto-fill of slots with entities of the same name.
+The `auto_fill` key in the domain is no longer available, as well as the `auto_fill` parameter in the constructor of
+the `Slot` class.
+
+While forms continue to request the next slot, slot extraction is now delegated to the default
+action `action_extract_slots`. This action runs in the background automatically after each user turn.
+Like `action_listen`, it should not be included in stories.
+
+Each slot in the `slots` section must include the `mappings` key. The same keys used for predefined mappings in 2.0 are
+available in 3.0. Additionally, you can define slots with custom mappings implemented in a custom action, for example:
+
+```yaml
+slots:
+  is_existing_customer:
+    type: bool
+    mappings:
+    - type: custom
+      action: action_verify_customer_status
+```
+
+You can also use [slot validation actions](./slot-validation-actions.mdx) to define custom validation of slots with
+predefined mappings or custom extraction and validation of slots with custom mappings.
+
+:::note
+The `required_slots` of a form has been converted in a list of slot names because slot mappings are now relocated
+from `forms` to `slots` in the domain.
```

Comment:
```suggestion
The `required_slots` of a form used to be a list of slot mappings.
Since slot mappings are relocated to the `slots` section of the domain, `required_slots` has been converted to a list of slot names only.```